### ISSUE TITLE
Holoviews: AttributeError: 'Array' object has no attribute 'flags'

### ISSUE THREAD

**Issue by jhamman** [pangeo-data/pangeo-tutorial-agu-2018#18]
Our `ocean_ssh_example` is raising the following error. Does anyone (@rabernat, @jbednar) have any ideas what's going on here?

[![Binder](https://binder.pangeo.io/badge_logo.svg)](https://binder.pangeo.io/v2/gh/pangeo-data/pangeo-tutorial/csdms2019)

Versions:
- holoviews                 1.12.2                     py_0    conda-forge
- hvplot                    0.4.0                      py_1    conda-forge
- numpy                     1.16.3           py36he5ce36f_0    conda-forge
- pandas                    0.24.2           py36hf484d3e_0    conda-forge
- xarray                    0.12.1                     py_0    conda-forge

```python-traceback
WARNING:param.load_subset: Callable raised "AttributeError("'Array' object has no attribute 'flags'",)".
Invoked as load_subset(numpy.datetime64('1993-01-01T00:00:00.000000000'))
WARNING:param.dynamic_operation: Callable raised "AttributeError("'Array' object has no attribute 'flags'",)".
Invoked as dynamic_operation(numpy.datetime64('1993-01-01T00:00:00.000000000'))
WARNING:param.dynamic_operation: Callable raised "AttributeError("'Array' object has no attribute 'flags'",)".
Invoked as dynamic_operation(numpy.datetime64('1993-01-01T00:00:00.000000000'))
WARNING:param.dynamic_operation: Callable raised "AttributeError("'Array' object has no attribute 'flags'",)".
Invoked as dynamic_operation(numpy.datetime64('1993-01-01T00:00:00.000000000'), height=400, scale=1.0, width=400, x_range=None, y_range=None)
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
/srv/conda/envs/notebook/lib/python3.6/site-packages/IPython/core/formatters.py in __call__(self, obj, include, exclude)
    968 
    969             if method is not None:
--> 970                 return method(include=include, exclude=exclude)
    971             return None
    972         else:

/srv/conda/envs/notebook/lib/python3.6/site-packages/holoviews/core/dimension.py in _repr_mimebundle_(self, include, exclude)
   1291         combined and returned.
   1292         """
-> 1293         return Store.render(self)
   1294 
   1295 

/srv/conda/envs/notebook/lib/python3.6/site-packages/holoviews/core/options.py in render(cls, obj)
   1360         data, metadata = {}, {}
   1361         for hook in hooks:
-> 1362             ret = hook(obj)
   1363             if ret is None:
   1364                 continue

/srv/conda/envs/notebook/lib/python3.6/site-packages/holoviews/ipython/display_hooks.py in pprint_display(obj)
    276     if not ip.display_formatter.formatters['text/plain'].pprint:
    277         return None
--> 278     return display(obj, raw_output=True)
    279 
    280 

/srv/conda/envs/notebook/lib/python3.6/site-packages/holoviews/ipython/display_hooks.py in display(obj, raw_output, **kwargs)
    252     elif isinstance(obj, (HoloMap, DynamicMap)):
    253         with option_state(obj):
--> 254             output = map_display(obj)
    255     elif isinstance(obj, Plot):
    256         output = render(obj)

/srv/conda/envs/notebook/lib/python3.6/site-packages/holoviews/ipython/display_hooks.py in wrapped(element)
    141         try:
    142             max_frames = OutputSettings.options['max_frames']
--> 143             mimebundle = fn(element, max_frames=max_frames)
    144             if mimebundle is None:
    145                 return {}, {}

/srv/conda/envs/notebook/lib/python3.6/site-packages/holoviews/ipython/display_hooks.py in map_display(vmap, max_frames)
    201         return None
    202 
--> 203     return render(vmap)
    204 
    205 

/srv/conda/envs/notebook/lib/python3.6/site-packages/holoviews/ipython/display_hooks.py in render(obj, **kwargs)
     64         renderer = renderer.instance(fig='png')
     65 
---> 66     return renderer.components(obj, **kwargs)
     67 
     68 

/srv/conda/envs/notebook/lib/python3.6/site-packages/holoviews/plotting/bokeh/renderer.py in components(self, obj, fmt, comm, **kwargs)
    248         # Bokeh has to handle comms directly in <0.12.15
    249         comm = False if bokeh_version < '0.12.15' else comm
--> 250         return super(BokehRenderer, self).components(obj,fmt, comm, **kwargs)
    251 
    252 

/srv/conda/envs/notebook/lib/python3.6/site-packages/holoviews/plotting/renderer.py in components(self, obj, fmt, comm, **kwargs)
    319             plot = obj
    320         else:
--> 321             plot, fmt = self._validate(obj, fmt)
    322 
    323         data, metadata = {}, {}

/srv/conda/envs/notebook/lib/python3.6/site-packages/holoviews/plotting/renderer.py in _validate(self, obj, fmt, **kwargs)
    218         if isinstance(obj, tuple(self.widgets.values())):
    219             return obj, 'html'
--> 220         plot = self.get_plot(obj, renderer=self, **kwargs)
    221 
    222         fig_formats = self.mode_formats['fig'][self.mode]

/srv/conda/envs/notebook/lib/python3.6/site-packages/holoviews/plotting/bokeh/renderer.py in get_plot(self_or_cls, obj, doc, renderer, **kwargs)
    133             curdoc().theme = self_or_cls.theme
    134         doc.theme = self_or_cls.theme
--> 135         plot = super(BokehRenderer, self_or_cls).get_plot(obj, renderer, **kwargs)
    136         plot.document = doc
    137         return plot

/srv/conda/envs/notebook/lib/python3.6/site-packages/holoviews/plotting/renderer.py in get_plot(self_or_cls, obj, renderer, **kwargs)
    184 
    185         # Initialize DynamicMaps with first data item
--> 186         initialize_dynamic(obj)
    187 
    188         if not isinstance(obj, Plot):

/srv/conda/envs/notebook/lib/python3.6/site-packages/holoviews/plotting/util.py in initialize_dynamic(obj)
    248             continue
    249         if not len(dmap):
--> 250             dmap[dmap._initial_key()]
    251 
    252 

/srv/conda/envs/notebook/lib/python3.6/site-packages/holoviews/core/spaces.py in __getitem__(self, key)
   1323         # Not a cross product and nothing cached so compute element.
   1324         if cache is not None: return cache
-> 1325         val = self._execute_callback(*tuple_key)
   1326         if data_slice:
   1327             val = self._dataslice(val, data_slice)

/srv/conda/envs/notebook/lib/python3.6/site-packages/holoviews/core/spaces.py in _execute_callback(self, *args)
   1096 
   1097         with dynamicmap_memoization(self.callback, self.streams):
-> 1098             retval = self.callback(*args, **kwargs)
   1099         return self._style(retval)
   1100 

/srv/conda/envs/notebook/lib/python3.6/site-packages/holoviews/core/spaces.py in __call__(self, *args, **kwargs)
    731 
    732         try:
--> 733             ret = self.callable(*args, **kwargs)
    734         except KeyError:
    735             # KeyError is caught separately because it is used to signal

/srv/conda/envs/notebook/lib/python3.6/site-packages/holoviews/util/__init__.py in dynamic_operation(*key, **kwargs)
    932             def dynamic_operation(*key, **kwargs):
    933                 kwargs = dict(self._eval_kwargs(), **kwargs)
--> 934                 return self._process(map_obj[key], key, kwargs)
    935         if isinstance(self.p.operation, Operation):
    936             return OperationCallable(dynamic_operation, inputs=[map_obj],

/srv/conda/envs/notebook/lib/python3.6/site-packages/holoviews/core/spaces.py in __getitem__(self, key)
   1323         # Not a cross product and nothing cached so compute element.
   1324         if cache is not None: return cache
-> 1325         val = self._execute_callback(*tuple_key)
   1326         if data_slice:
   1327             val = self._dataslice(val, data_slice)

/srv/conda/envs/notebook/lib/python3.6/site-packages/holoviews/core/spaces.py in _execute_callback(self, *args)
   1096 
   1097         with dynamicmap_memoization(self.callback, self.streams):
-> 1098             retval = self.callback(*args, **kwargs)
   1099         return self._style(retval)
   1100 

/srv/conda/envs/notebook/lib/python3.6/site-packages/holoviews/core/spaces.py in __call__(self, *args, **kwargs)
    731 
    732         try:
--> 733             ret = self.callable(*args, **kwargs)
    734         except KeyError:
    735             # KeyError is caught separately because it is used to signal

/srv/conda/envs/notebook/lib/python3.6/site-packages/holoviews/util/__init__.py in dynamic_operation(*key, **kwargs)
    932             def dynamic_operation(*key, **kwargs):
    933                 kwargs = dict(self._eval_kwargs(), **kwargs)
--> 934                 return self._process(map_obj[key], key, kwargs)
    935         if isinstance(self.p.operation, Operation):
    936             return OperationCallable(dynamic_operation, inputs=[map_obj],

/srv/conda/envs/notebook/lib/python3.6/site-packages/holoviews/core/spaces.py in __getitem__(self, key)
   1323         # Not a cross product and nothing cached so compute element.
   1324         if cache is not None: return cache
-> 1325         val = self._execute_callback(*tuple_key)
   1326         if data_slice:
   1327             val = self._dataslice(val, data_slice)

/srv/conda/envs/notebook/lib/python3.6/site-packages/holoviews/core/spaces.py in _execute_callback(self, *args)
   1096 
   1097         with dynamicmap_memoization(self.callback, self.streams):
-> 1098             retval = self.callback(*args, **kwargs)
   1099         return self._style(retval)
   1100 

/srv/conda/envs/notebook/lib/python3.6/site-packages/holoviews/core/spaces.py in __call__(self, *args, **kwargs)
    731 
    732         try:
--> 733             ret = self.callable(*args, **kwargs)
    734         except KeyError:
    735             # KeyError is caught separately because it is used to signal

/srv/conda/envs/notebook/lib/python3.6/site-packages/holoviews/util/__init__.py in dynamic_operation(*key, **kwargs)
    932             def dynamic_operation(*key, **kwargs):
    933                 kwargs = dict(self._eval_kwargs(), **kwargs)
--> 934                 return self._process(map_obj[key], key, kwargs)
    935         if isinstance(self.p.operation, Operation):
    936             return OperationCallable(dynamic_operation, inputs=[map_obj],

/srv/conda/envs/notebook/lib/python3.6/site-packages/holoviews/core/spaces.py in __getitem__(self, key)
   1323         # Not a cross product and nothing cached so compute element.
   1324         if cache is not None: return cache
-> 1325         val = self._execute_callback(*tuple_key)
   1326         if data_slice:
   1327             val = self._dataslice(val, data_slice)

/srv/conda/envs/notebook/lib/python3.6/site-packages/holoviews/core/spaces.py in _execute_callback(self, *args)
   1096 
   1097         with dynamicmap_memoization(self.callback, self.streams):
-> 1098             retval = self.callback(*args, **kwargs)
   1099         return self._style(retval)
   1100 

/srv/conda/envs/notebook/lib/python3.6/site-packages/holoviews/core/spaces.py in __call__(self, *args, **kwargs)
    731 
    732         try:
--> 733             ret = self.callable(*args, **kwargs)
    734         except KeyError:
    735             # KeyError is caught separately because it is used to signal

/srv/conda/envs/notebook/lib/python3.6/site-packages/holoviews/core/data/__init__.py in load_subset(*args)
    720             def load_subset(*args):
    721                 constraint = dict(zip(dim_names, args))
--> 722                 group = self.select(**constraint)
    723                 if np.isscalar(group):
    724                     return group_type(([group],), group=self.group,

/srv/conda/envs/notebook/lib/python3.6/site-packages/holoviews/core/data/__init__.py in select(self, selection_specs, **selection)
    382             return self
    383 
--> 384         data = self.interface.select(self, **selection)
    385 
    386         if np.isscalar(data):

/srv/conda/envs/notebook/lib/python3.6/site-packages/holoviews/core/data/xarray.py in select(cls, dataset, selection_mask, **selection)
    460             # we can monkeypatch the dataarray back to writeable.
    461             for d in data.values():
--> 462                 d.data.flags.writeable = True
    463 
    464         da = dask_array_module()

AttributeError: 'Array' object has no attribute 'flags'
```

**Comment by jhamman** [comment]
btw, this is also popping up in the landsat notebook.

**Comment by jbednar** [comment]
Should be fixed in holoviews 1.12.3.

**Comment by jhamman** [comment]
Thanks @jbednar!
